app/models/patient.rb

```diff
@@ -215,11 +216,19 @@ def search(name_or_phone_str, lines = LINES)
       all_matching_names = find_name_matches name_regexp, lines
       all_matching_phones = find_phone_matches phone_regexp, lines
 
-      (all_matching_names | all_matching_phones)
+      sort_and_limit_patient_matches(all_matching_names, all_matching_phones)
     end
 
     private
 
+
+    def sort_and_limit_patient_matches(all_matching_names, all_matching_phones)
+      matches = (all_matching_names | all_matching_phones)
+      matches.sort{|a,b|
```

Comment:
This'll work very nicely, I think it's great!

The world's most minor style suggestion: The ruby style for blocks that span one line is  `{ |x| ... }`, and for blocks with line breaks it's `do |x|`. Just so it's a little more 'ruby' stylistically,  you might want to toss the block on one line. NBD if not though, up to you!